Find Poco library using upstream CMake configuration.

nuclearsandwich opened the pull request:
A bug in the FindPoco module from cmake_modules causes incorrect linking in Debug configurations.

I haven't decided if this is a straw man or not since I'm not sure what we gain or lose by not using the module from cmake_modules.

This is one possible way to resolve #117 but I'd like input from someone like @wjwwood before I decide whether this is a good idea or not.

nuclearsandwich committed:
Find Poco library using upstream CMake configuration.

A bug in the FindPoco module from cmake_modules causes incorrect linking
in Debug configurations.

I haven't decided if this is a straw man or not since I'm not sure what
we gain or lose by not using the module from cmake_modules.

dirk-thomas commented on CMakeLists.txt:
```diff
@@ -19,7 +19,6 @@ endif()
 find_package(console_bridge REQUIRED)
 
 if(${catkin_FOUND})
-  find_package(catkin REQUIRED COMPONENTS cmake_modules)
```
The dependency on `cmake_modules` should then be removed from the manifest.

nuclearsandwich commented:
CI informs me that this won't be a complete solution as Debian Stretch's libpoco-dev package doesn't contain CMake configuration.

wjwwood commented:
+1 to getting rid of stuff in `cmake_modules` where ever possible, but as you said, we likely created it there to address a specific problem like debian not shipping proper CMake module or config file.

nuclearsandwich committed:
Drop cmake_modules dependency.

We were using this only for Poco. As of Debian Buster and Ubuntu Bionic
libpoco-dev provides the necessary CMake configuration.

nuclearsandwich commented:
While this would be the preferred solution to #117 it won't work for Melodic. I've added a commit that completes this PR so it can be considered in the future once support for Stretch is dropped.
